systemtest/benchtest/main.go

```diff
@@ -265,6 +256,9 @@ func warmup(agents int, events uint, url, token string) error {
 	var merr multierror.Errors
 	for err := range errs {
 		if err != nil && !errors.Is(err, context.DeadlineExceeded) {
+			if e, ok := err.(timeoutError); ok && e.Timeout() {
```

Comment:
**suggestion:** I think we can use `errors.As` and `net.Error` to check if `err` is a network error and then call `Timeout()` on it.
We should also add a TODO to remove this once we update to Go 1.19 as the workaround is no longer necessary: https://go.dev/doc/go1.19#net